src/textual/app.py

```diff
@@ -1321,6 +1337,14 @@ def push_screen(self, screen: Screen | str) -> AwaitMount:
             screen: A Screen instance or the name of an installed screen.
 
         """
+        if not isinstance(screen, (Screen, str)):
+            raise TypeError(
+                f"push_screen requires a Screen instance or str; not {screen!r}"
+            )
+
+        if self._screen_stack:
+            self.screen.post_message(events.ScreenSuspend())
+            self.screen.refresh()
         next_screen, await_mount = self._get_screen(screen)
         self._screen_stack.append(next_screen)
         self.screen.post_message(events.ScreenResume())
```

Comment:
By appending to `_screen_stack`, then `self.screen` will already point to `next_screen`?
I think it would be clearer if this was
```suggestion
        self._screen_stack.append(next_screen)
        next_screen.post_message(events.ScreenResume())
```

Make it easier to follow to what screen the message is being posted.